sql-plugin/src/main/scala/com/nvidia/spark/rapids/ParquetCachedBatchSerializer.scala

```diff
@@ -860,11 +869,19 @@ protected class ParquetCachedBatchSerializer extends GpuCachedBatchSerializer {
         if (!cbIter.hasNext) {
           Iterator.empty
         } else {
-          new ShimCurrentBatchIterator(
-            cbIter.next().asInstanceOf[ParquetCachedBatch],
-            conf,
-            selectedAttributes,
-            options, hadoopConf)
+          val cachedBatch = cbIter.next().asInstanceOf[ParquetCachedBatch]
+          try {
+            new ShimCurrentBatchIterator(
+              cachedBatch,
+              conf,
+              selectedAttributes,
+              options, hadoopConf)
+          } catch {
+            case r: RuntimeException if r.getMessage.contains("too small length: 0") =>
```

Comment:
Please don't use an exception to trigger this use case, and especially don't inspect the exception message. That is super brittle. You have the `cachedBatch` just a few line above here. Please look at it before sending it to the `ShimCurrentBatchIterator`.

Reply:
Yeah, I realized that after I pushed this. I will update right after my tests pass